docs/sources/flow/reference/components/loki.source.file.md

```diff
@@ -26,19 +26,44 @@ log entries to the list of receivers passed in `forward_to`.
 
 `loki.source.file` supports the following arguments:
 
-Name         | Type                   | Description          | Default | Required
------------- | ---------------------- | -------------------- | ------- | --------
-`targets`    | `list(map(string))`    | List of files to read from. | | yes
-`forward_to` | `list(LogsReceiver)`   | List of receivers to send log entries to. | | yes
-`encoding`   | `string`               | The encoding to convert from when reading files. | `""` | no
+ Name         | Type                 | Description                                      | Default | Required 
+--------------|----------------------|--------------------------------------------------|---------|----------
+ `targets`    | `list(map(string))`  | List of files to read from.                      |         | yes      
+ `forward_to` | `list(LogsReceiver)` | List of receivers to send log entries to.        |         | yes      
+ `encoding`   | `string`             | The encoding to convert from when reading files. | `""`    | no       
+
+The `encoding` argument must be a valid [IANA encoding][] name. If not set, it
+defaults to UTF-8.
 
 ## Blocks
 
-The `loki.source.file` component doesn't support any inner blocks and is
-configured fully through arguments.
+The following blocks are supported inside the definition of `loki.source.file`:
 
-The `encoding` argument must be a valid [IANA encoding][] name. If not set, it
-defaults to UTF-8. 
+ Hierarchy      | Name               | Description                                   | Required 
+----------------|--------------------|-----------------------------------------------|----------
+ decompresssion | [decompresssion][] | Configure reading logs from compressed files. | no       
+
+[decompresssion]: #decompresssion-block
+
+### decompresssion block
+
+The `decompression` block contains configuration for reading logs from 
+compressed files. The following arguments are supported:
```

Comment:
```suggestion
The `decompression` block contains configuration for reading logs from 
compressed files found in its `targets`. The following arguments are supported:
```

Possible addition so that it doesn't read that if this is enabled, the component _only_ reads compressed files. WDYT? Any better wording?

Reply:
> so that it doesn't read that if this is enabled, the component only reads compressed files

The component indeed will _only read from compressed files_, trying to decompress all the targets. I think it could be improved, but I wouldn't want to divert from promtails implementation right now.

Reply:
Oh, I see. I'm not particularly fond of that decision, but yeah, let's keep in line with Promtail for now.